lib/ts/framework/utils.ts

```diff
@@ -23,7 +22,82 @@ import type { HTTPMethod } from "../types";
 import { COOKIE_HEADER } from "./constants";
 import { getFromObjectCaseInsensitive } from "../utils";
 import contentType from "content-type";
-import inflate from "inflation";
+import pako from "pako";
+
+let brotliDecompress: ((input: Buffer) => Buffer) | null = null;
+
+try {
+    // @ts-ignore
+    if (typeof EdgeRuntime === "undefined") brotliDecompress = require("brotli").decompress;
+} catch (error) {
+    brotliDecompress = null;
+}
+
+async function inflate(stream: IncomingMessage): Promise<string> {
+    if (!stream) {
+        throw new TypeError("argument stream is required");
+    }
+
+    const encoding = (stream.headers && stream.headers["content-encoding"]) || "identity";
+
+    let decompressedData: Uint8Array | string;
+
+    if (encoding === "gzip" || encoding === "deflate") {
+        const inflator = new pako.Inflate();
+
+        for await (const chunk of stream) {
+            inflator.push(chunk, false);
+        }
+
+        if (inflator.err) {
+            throw new Error(`Decompression error: ${inflator.msg}`);
+        }
+
+        decompressedData = inflator.result;
+    } else if (encoding === "br") {
+        if (!brotliDecompress) throw new Error("Brotli decompression not supported on the platform");
```

Comment:
How does this show up? Does it crash the server or produces a 500 response?

Reply:
It doesn't crash the server but a BAD_INPUT_ERROR is returned. This is handled by the parent function and thrown.

I saw however, that the parent function did not care about the type of error so I did not update it.

Here's the line I am referring to: https://github.com/supertokens/supertokens-node/blob/f51a72cb23490a95b3e69811e16a4cd2684589ac/lib/ts/framework/utils.ts#L182

Reply:
I see, but then `Please make sure to pass a valid JSON input in the request body` is not the correct error message.

Reply:
I agree but since that was going to require more changes that were not directly related to this change, I thought it was out of scope.

Let me know if I should go ahead and make those changes.

Reply:
Updated the code as per our discussion.